include/circt/Dialect/APInt/APInt.td

```diff
@@ -0,0 +1,41 @@
+//===- APInt.td - APInt dialect definition -----------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is the top level file for the APInt dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef APINT_TD
+#define APINT_TD
+
+include "mlir/IR/OpBase.td"
+
+def APIntDialect : Dialect {
+  let name = "apint";
+
+  let summary = "Types and operations for apint dialect";
+  let description = [{
+    This dialect defines the `apint` dialect, which is intended to represent
+    lossless arbitrary precision integer arithmetics.
```

Comment:
A little more verbosity would be useful here.  In what sense are the operations "lossless"? :)

Reply:
I will try my best :)